jacodb-panda-dynamic/src/main/kotlin/org/jacodb/panda/dynamic/ets/dto/convert.kt

```diff
@@ -0,0 +1,712 @@
+                args = value.args.map {
+                    val etsEntity = convertToEtsEntity(it)
+                    if (etsEntity is EtsValue) return@map etsEntity
+                    val newLocal = EtsLocal("_tmp${freeLocal++}", EtsUnknownType)
+                    currentStmts += EtsAssignStmt(
+                        location = loc(),
+                        lhv = newLocal,
+                        rhv = etsEntity,
+                    )
+                    newLocal
+                },
+            )
+
+            is ThisRefDto -> EtsThis(
+                type = convertToEtsType(value.type) // TODO: as ClassType
+            )
+
+            is ParameterRefDto -> EtsParameterRef(
+                index = value.index,
+                type = convertToEtsType(value.type),
+            )
+
+            is ArrayRefDto -> EtsArrayAccess(
+                array = convertToEtsEntity(value.array),
+                index = convertToEtsEntity(value.index),
+                type = convertToEtsType(value.type),
+            )
+
+            is FieldRefDto -> convertToEtsFieldRef(value)
+
+            // else -> error("Unknown Value: $value")
+        }
+    }
+
+    fun convertToEtsFieldRef(fieldRef: FieldRefDto): EtsFieldRef {
+        val field = convertToEtsFieldSignature(fieldRef.field)
+        return when (fieldRef) {
+            is InstanceFieldRefDto -> EtsInstanceFieldRef(
+                instance = convertToEtsEntity(fieldRef.instance), // as Local
+                field = field
+            )
+
+            is StaticFieldRefDto -> EtsStaticFieldRef(
+                field = field
+            )
+        }
+    }
+
+    fun cfg2cfg(cfg: CfgDto): EtsCfg {
+        require(cfg.blocks.isNotEmpty()) {
+            "Method body should contain at least return stmt"
+        }
+
+        val visited: MutableSet<Int> = hashSetOf()
+        val queue: ArrayDeque<Int> = ArrayDeque()
+        queue.add(0)
+
+        val blocks = cfg.blocks.associateBy { it.id }
+        val blockStart: MutableMap<Int, Int> = hashMapOf()
+        val blockEnd: MutableMap<Int, Int> = hashMapOf()
+
+        while (queue.isNotEmpty()) {
+            val block = blocks[queue.removeFirst()]!!
+            if (block.stmts.isNotEmpty()) {
+                blockStart[block.id] = currentStmts.size
+            }
+            for (stmt in block.stmts) {
+                convertToEtsStmt(stmt)
+            }
+            if (block.stmts.isNotEmpty()) {
+                blockEnd[block.id] = currentStmts.lastIndex
+            }
+            for (next in block.successors) {
+                if (visited.add(next)) {
+                    queue.addLast(next)
+                }
+            }
+        }
+
+        val successorMap: MutableMap<EtsStmt, List<EtsStmt>> = hashMapOf()
+        for (block in cfg.blocks) {
+            if (block.stmts.isEmpty()) {
+                continue
+            }
+            val startId = blockStart[block.id]!!
```

Comment:
getValue insted of !!

Reply:
What's the difference here? `!!` is more concise. And we guarantee it cannot fail here, just by the consturction.